src/transform-value.js

```diff
@@ -15,29 +15,13 @@
 (function(internal, scope) {
 
   function TransformValue(values) {
-    if (!Array.isArray(values)) {
-      throw new TypeError('TransformValue must be an array of ' +
-          'TransformComponents.');
-    } else if (values.length < 1) {
-      throw new TypeError('TransformValue must have at least 1 ' +
-          'TransformComponent.');
+    if (values === undefined) {
+      return this._createEmptyTransform();
     }
-
-    this.transformComponents = [];
-    for (var i = 0; i < values.length; i++) {
-      if (!(values[i] instanceof TransformComponent)) {
-        throw new TypeError('Argument at index ' + i + ' is not an instance ' +
-            'of \'TransformComponent\'.');
-      }
-      this.transformComponents.push(values[i]);
-    }
-
-    this._matrix = this._computeMatrix();
-    this.cssString = this._generateCssString();
+    return this._createSequenceTransform(values);
```

Comment:
These functions don't return anything, so this looks a bit strange. Also there's a lot of indirection in the constructor that jumps all around the file now. It's also strange that no argument is treated differently to passing in an empty sequence. Maybe @shans should clarify what the expected behaviour for these cases is? It is not currently covered in the spec.

It would be clearer to keep the setting of _matrix and cssString in the constructor, but  modify the computeMatrix function to look like this:

```javascript
  function TransformValue(values) {
    if (values === undefined) {
      values = [];
    }
    if (!Array.isArray(values)) {
      throw ...
    }
// Keep the old version of the
  }
...

  TransformValue.prototype._computeMatrix = function() {
    if (!this.transformComponents.length) {
      // return identity matrix
    }
    var matrix = this.transformComponents[0].asMatrix();
    for (var i = 1; i < this.transformComponents.length; ++i) {
      matrix = matrix.multiply(this.transformComponents[i].asMatrix());
    }
    return matrix;
  };
```